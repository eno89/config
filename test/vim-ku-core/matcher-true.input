" Test: vim-ku-core matcher-true
function s:describe__ku_matcher_true_matches_p()  "{{{1
  let args = {'pattern': 'rail'}

  It should return true regardless of candidate nor args

  Should !!ku#matcher#true#matches_p({'word': 'rail'}, args) == 1
  Should !!ku#matcher#true#matches_p({'word': 'a rail'}, args) == 1
  Should !!ku#matcher#true#matches_p({'word': 'rail gun'}, args) == 1
  Should !!ku#matcher#true#matches_p({'word': 'a rail gun'}, args) == 1
  Should !!ku#matcher#true#matches_p({'word': 'arailgun'}, args) == 1

  Should !!ku#matcher#true#matches_p({'word': ''}, args) == 1
  Should !!ku#matcher#true#matches_p({'word': 'reil gan'}, args) == 1
  Should !!ku#matcher#true#matches_p({'word': 'RAILgun'}, args) == 1

  Should !!ku#matcher#true#matches_p([], []) == 1
endfunction




" __END__  "{{{1
" vim: filetype=vim foldmethod=marker
