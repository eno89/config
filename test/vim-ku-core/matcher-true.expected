==== ku_matcher_true_matches_p 
---- It should return true regardless of candidate nor args 
.........



**** Result ****


9 examples, 0 failures
